Fix generation of contribution page

Modern multimarkdown tends to convert the <% and %> tags into HTML when they
occur before or after an empty line. This would then break parsing with erb.
Remove all empty lines in the vincinity of these tags to prevent this problem.

Additionally, move the cache file for the githubcredits into the build
directory, because the home is either not writable, or not preserved within
MacPorts builds, and attempting to re-generate these texts too often usually
causes one to hit the GitHub API limit. By putting the file into the build
directory, aborted builds can at least be re-started without hitting any API
limits.

Upstream-Status: Inappropriate [configuration]
--- Applications/TextMate/about/Contributions.md.orig   2018-01-31 21:46:35.000000000 +0100
+++ Applications/TextMate/about/Contributions.md        2018-01-31 22:21:35.000000000 +0100
@@ -8,13 +8,11 @@

 <div>
 <%# this wrapping div prevents Markdown from trying to parse the ERB blocks %>
-
 <%
 last_group_heading = ''
 require File.join(File.dirname(__FILE__), 'bin/gen_credits')
-generate_credits(File.expand_path('~/Library/Caches/com.macromates.TextMate/githubcredits'), warn) do |hash, author, subject, body, userpic, date, github_user|
+generate_credits(File.expand_path('githubcredits'), warn) do |hash, author, subject, body, userpic, date, github_user|
   group_heading = date.strftime('%b %e, %Y')
-
   if last_group_heading != group_heading
     if last_group_heading != ''
         _erbout << "</ol>\n"
@@ -49,7 +47,6 @@
   </li>
 <%
 end
-
 if last_group_heading != ''
     _erbout << "</ol>\n"
 end
